sonar-php-plugin/src/main/java/org/sonar/plugins/php/PHPSensor.java

```diff
@@ -132,17 +133,27 @@ public void analyse(Project project, SensorContext context) {
 
     for (InputFile inputFile : inputFiles) {
       progressReport.nextFile();
-      phpAnalyzer.nextFile(inputFile.file());
-
-      saveIssues(phpAnalyzer.analyze(), inputFile);
-      saveHighlighting(phpAnalyzer.getHighlighting(), inputFile);
-      saveNewFileMeasures(phpAnalyzer.computeMeasures(fileLinesContextFactory.createFor(inputFile)), inputFile);
+      analyseFile(phpAnalyzer, inputFile);
     }
 
     progressReport.stop();
 
   }
 
+  private void analyseFile(PHPAnalyzer phpAnalyzer, InputFile inputFile) {
+    try {
+      phpAnalyzer.nextFile(inputFile.file());
+    } catch (RecognitionException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)